docker/pytorch/Dockerfile

```diff
@@ -162,9 +162,6 @@ RUN useradd -rm -d /home/mosaicml -s /bin/bash -u 1000 -U -s /bin/bash mosaicml
 
 WORKDIR /home/mosaicml
 
-USER mosaicml
```

Comment:
[nit] Can we also just get rid of the `useradd` for `mosaicml` user above?

Reply:
I'd like to keep this user around for now, should someone want to run the docker image not as root (e.g. so permissions don't get screwed up if using a volume mount). But based off the PR discussion, agree it does make sense to default to root.